packages/elements/src/internals/machines/sign-in.machine.ts

```diff
@@ -144,77 +151,145 @@ export const SignInMachine = setup({
         },
       }),
     },
-    'OAUTH.CALLBACK': '.SSOCallbackRunning',
+    'OAUTH.CALLBACK': '#SignIn.SSOCallbackRunning',
   },
+  initial: 'DeterminingState',
   states: {
-    Init: {
+    DeterminingState: {
+      initial: 'Init',
+      states: {
+        Init: {
+          always: [
+            {
+              description: 'If the SignIn resource is empty, invoke the sign-in start flow.',
+              guard: 'isServer',
+              target: 'Server',
+            },
+            {
+              description: 'Wait for the Clerk instance to be ready.',
+              guard: 'isBrowser',
+              target: 'Browser',
+            },
+          ],
+        },
+        Server: {
+          description: 'Determines the state of the sign-in flow on the server. This is a no-op for now.', // TODO: Implement
+          entry: ['debug', log('Server no-op')],
+        },
+        Browser: {
+          description: 'Determines the state of the sign-in flow on the browser.',
+          always: [
+            {
+              description: 'Wait for the Clerk instance to be ready.',
+              guard: not('isClerkLoaded'),
+              target: '#SignIn.WaitingForClerk',
+            },
+            {
+              description: 'If loggedin and single-session, invoke the sign-in start flow with error.',
+              guard: and(['isLoggedIn', 'isSingleSessionMode']),
+              actions: assign({ error: () => new Error('Already logged in.') }),
+              target: '#SignIn.Start',
```

Comment:
Related to: https://linear.app/clerk/issue/SDK-1137/[sign-in]-handle-already-signed-in-error